### PR TITLE
Switch to MacKernelSDK

### PR DESCRIPTION
**Pull request by kprinssu**
This should've been done a while ago. This PR is to replace the macOS SDK by Apple with MacKernelSDK from acidanthera.

Should address https://github.com/VoodooI2C/VoodooI2C/issues/383

### CODE REVIEW

**Comment by kprinssu** [comment]
@ben9923 I am not sure if this actually builds against the MacKernelSDK binaries. 

@vit9696 can possible illuminate me here, but adding the MacKernelSDK to `LIBRARY_SEARCH_PATHS` should do the trick?

**Comment by kprinssu** [comment]
Additionally, we need to update the Satellite and Dependency projects as well.

**Comment by kprinssu** [comment]
MacKernelSDK now has the private headers from VoodooI2C.

https://github.com/acidanthera/MacKernelSDK/commit/ae18f528eb6d4d461751565dba041e4301c3b776

File: VoodooI2C/VoodooI2C/VoodooI2CDevice/VoodooI2CDeviceNub.cpp [hunk 1]
```diff
@@ -9,6 +9,12 @@
 #include "../VoodooI2CController/VoodooI2CControllerDriver.hpp"
 #include "VoodooI2CDeviceNub.hpp"
 
+#include <Availability.h>
+
+#ifndef __ACIDANTHERA_MAC_SDK
+#error "This kext SDK is unsupported. Dowload from https://github.com/acidanthera/MacKernelSDK"
```
Comment (zhen-zen) [inline VoodooI2C/VoodooI2C/VoodooI2CDevice/VoodooI2CDeviceNub.cpp:15]: Minor typo here

Comment (kprinssu) [reply VoodooI2C/VoodooI2C/VoodooI2CDevice/VoodooI2CDeviceNub.cpp:15]: It's meant to be a preprocessor error to ensure that the MacKernelSDK is present. 

TBH, I rather not have it as a global include and have it fail when attempting to build the implementation (cpp file). The error is quite straightforward and easy to understand.

Comment (vit9696) [reply VoodooI2C/VoodooI2C/VoodooI2CDevice/VoodooI2CDeviceNub.cpp:15]: Dowload → Download ^^

**Comment by kprinssu** [comment]
Thanks @ben9923 for majority of the work here.

**Pull request merged**